docs/developers/tracing.md

```diff
@@ -26,10 +26,12 @@ Check the official [Jaeger docs](https://www.jaegertracing.io/docs/) on how to w
 
 ## Enabling tracing
 
-Tekton pipelines controller expects the following environment variables to be able to connect to jaeger:
+The configmap `config/config-tracing.yaml` contains the configuration for tracing. It contains the following fields:
+
+* enabled: Set this to true to enable tracing
+* endpoint: API endpoint for jaeger collector to send the traces. By default the endpoint is configured to be `http://jaeger-collector.jaeger.svc.cluster.local:14268/api/traces`.
+
+Tekton pipelines controller also supports the following additional environment variables to be able to connect to jaeger:
 
-* `OTEL_EXPORTER_JAEGER_ENDPOINT` is the HTTP endpoint for sending spans directly to a collector.
 * `OTEL_EXPORTER_JAEGER_USER` is the username to be sent as authentication to the collector endpoint.
```

Comment:
I'm curious why some of the configuration is moving to a configmap but not all of it. Is it because the password can be read from a secret when used in an envvar but not when used in a configmap? maybe the configmap could accept the name of a secret where the password is stored?

Reply:
Yes that's the idea. Configmap will have a secret name where creds are stored. It requires a separate logic for reconciling the secret. I wanted to keep the scope of the PR small for easy review process. I already have a WIP follow-up PR for this change in my fork [here](https://github.com/kmjayadeep/pipeline/pull/4) if you are interested to take a look.